test/topology/test_snapshot.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""
+Test snapshot transfer by forcing threshold and performing schema changes
+"""
+import asyncio
+import logging
+from test.pylib.rest_client import inject_error_one_shot, inject_error
+import pytest
+from cassandra.query import SimpleStatement              # type: ignore # pylint: disable=no-name-in-module
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_snapshot(manager, random_tables):
+    """
+        Cluster A, B, C
+        with reduced snapshot threshold create table, do several schema changes, insert
+        start a new server, stop A B C, check if it sees the table schema correctly with an insert.
+    """
+    server_a, server_b, server_c = await manager.running_servers()
+    t = await random_tables.add_table(ncolumns=5, pks=1)
+
+    # Reduce the snapshot thresholds
+    errs = [inject_error_one_shot(manager.api, s.ip_addr, 'raft_server_snapshot_reduce_threshold')
+            for s in [server_a, server_b, server_c]]
+    await asyncio.gather(*errs)
+    # error injection is in Raft io_thread code path, wait for it to run
```

Comment:
Now I see what you meant. You mean `applier_fiber` (the other one is `io_fiber`, not `io_thread`, but that's also not where you put the injection; you put it in `applier_fiber`).

Let's not do this like that. The sleep doesn't guarantee anything and it's just not an elegant way to solve the problem.

I suggest you do an additional schema change here. Once the schema change returns you're guaranteed that the fiber loop executed at least once on the node where you requested the schema change, because we add entry with `wait_type::applied`. We could actually move the injection to `io_fiber` - then it will be guaranteed even if we add_entry with `wait_type::committed`. Whatever you pick, this should be explained in a comment.